### PR TITLE
rebuild wideresnet using reduce operator

### PR DESCRIPTION
**Pull request by brettkoonce**

### CODE REVIEW

File: CIFAR/wide_resnet.swift [hunk 1]
```diff
@@ -144,118 +133,84 @@ struct WideResNet16: Layer {
...
-        self.block4 = BatchNormConv2DBlock(
-            filterShape: (kernelSize, kernelSize,
-                featureCounts.1 * widenFactor, featureCounts.1 * widenFactor),
-            strides: (1, 1))
-        self.shortcut = Conv2D(
-            filterShape: (1, 1, featureCounts.0, featureCounts.1 * widenFactor),
-            strides: initialStride)
+extension WideResNet {
+    enum Kind {
+        case wideresnet16
```
Comment (rxwei) [inline CIFAR/wide_resnet.swift]: How about using camel case: `wideResNet16`?

For consistency, it's better to change `ResNet.Kind` to use camel case as well.

File: CIFAR/wide_resnet.swift [hunk 2]
```diff
@@ -47,93 +47,82 @@ struct BatchNormConv2DBlock: Layer {
     }
 }
 
-struct WideResnet16FirstBasicBlock: Layer {
+struct WideResnetBasicBlock: Layer {
```
Comment (rxwei) [inline CIFAR/wide_resnet.swift]:
```suggestion
struct WideResNetBasicBlock: Layer {
```

Comment (rxwei) [reply CIFAR/wide_resnet.swift]: s/Resnet/ResNet/g

Comment (brettkoonce) [reply CIFAR/wide_resnet.swift]: done!

File: CIFAR/wide_resnet.swift (hunk 1 above)
Comment (brettkoonce) [reply CIFAR/wide_resnet.swift]: reworked!

**Comment by rxwei** [comment]
Thanks!

**Comment by rxwei** [comment]
Hi @brettkoonce, this patch broke the SwiftPM build. Since we made all models use SwiftPM as the primary method of building, `swift run` is strongly preferred over running individual files as a script. Please help make sure everything builds and passes tests via SwiftPM. Thanks!

**Pull request merged**